### DISCUSSION TITLE
"Security, Privacy and Anonymity in Linux Mint" Guide.

### DISCUSSION THREAD

**Discussion by MikeNavy** [linuxmint/discussions#34] [Tutorials]
Hi,

I have written a PDF Guide: "Security, Privacy and Anonymity in Linux Mint".

You can download it and use it for free, it is distributed under Creative Commons license, Attribution-NonCommercial 4.0 International (CC BY-NC 4.0), see https://creativecommons.org/licenses/by-nc/4.0/.

Download link from my personal website (hosted by IONOS, no ad, no malware):
https://nallino.net/stockage/security/Linux_Mint_Security.pdf

It is also available from archive.org:
_[new revisions may appear a bit later on archive.org]_
https://archive.org/details/linux-mint-security

Privacy: the PDF document can be downloaded using Tor Browser.

Here is the Table of Contents:

**Introduction**
**Ubuntu Main Security Features**
**Threats List**
Local security attacks, needing physical access to computer
Distant security attacks
Various security threats
Privacy threats
Anonymity threat
**Prevention**
Protect the access to your computer
Update your system
Increase your system intrinsic security with Ubuntu Pro
Use trusted sources
Use a firewall
Sandbox your applications
Safe browsing
Be careful with downloaded files or attachments
Don't use Wine or Mono to run Windows programs
Set your system security
Reduce what your ISP can know
Protect your mails
Protect yourself from spam
Install LanguageTool local server
Use local translation programs
Avoid to have your personal data stolen
Stay anonymous
Protect your LAN against wireless intrusions
Protect yourself against Social Engineering
**Detection**
Malware and viruses detection
Intrusion detection
**Pre-Established Arrangements**
Elaborate a recover strategy
Backup and restore strategy
Proposed minimum backup and restore strategy
**Annex 1: Launching Commands and GUI Applications with Superuser Rights
Annex 2: Password Protect your GRUB Menu
Annex 3: Password Selection
Annex 4: Encryption
Annex 5: How to Enable Ubuntu Pro on Linux Mint
Annex 6: Flatpak Tutorial
Annex 7: Multiboot
Annex 8: Mullvad Flatpak on Tor Network, a Secure Alternative to Tor Browser
Annex 9: Tripwire Tutorial
Annex 10: Install and Set Up Free Proton VPN
Annex 11: Enhancing Firefox Security and Privacy
Annex 12: On-Demand Scan of Confidential Files
Annex 13: Internet Connection Diagrams
Threats / Prevention means correspondence matrix**

Feel free to comment, to improve next revision.

The guide is periodically updated and completed. See https://forums.linuxmint.com/viewtopic.php?p=2334665#p2334665 for the list of changes.

Regards,

MN

PS: The goal of this discussion is to show that Linux Mint security can be improved, and to explain why. Since most users will not have the capability / time / will to implement such changes, Linux Mint might implement some by default.

_[Latest edited on 2025-07-19]_

**Comment by bryce-carson** [comment]
@MikeNavy the links in your URLs are incorrect. The text of the link is correct, but all three links URL is `https://github.com/orgs/linuxmint/discussions/url`, except the link to VirusTotal.

**Comment by MikeNavy** [comment]
> @MikeNavy the links in your URLs are incorrect. The text of the link is correct, but all three links URL is `https://github.com/orgs/linuxmint/discussions/url`, except the link to VirusTotal.
Hi,

I have corrected links; the problem comes from the "Add a link" tool, when used all the links point to GitHub. I have simply copied the links without formatting.
Regards,
MN

**Comment by AtomicRobotMan0101** [comment]
4.12 Perhaps make mention that DoH can also be used in Thunderbird.

4.13.4 DuckDuckGo and Firefox also have Private Email Address redirection services.

Why not put this on Reddit as well? r/linuxmint  ....

**Reply by MikeNavy** [reply]
@AtomicRobotMan0101 
Hi,

Thanks for your comments.

1) DNS over HTTPS for Thunderbird is treated in 4.11.3 "Use DNS over HTTPS protocol".

2) DuckDuckGo mail redirection service: it may be interesting since the mails are transferred to your real mail address with mail trackers blocking. It is not anonymous mail, since DuckDuckGo knows your real e-mail address. **But mail redirection service implies some privacy degradation: your mails are read by DuckDuckGo servers**. (There is an extension with a similar functioning, vault56 protection, that submit your mails to its servers to remove trackers). 
_[I prefer the use of uBlock Origin in Thunderbird: filtering is done locally, on your computer, without sending your mails to a server. I have mentioned it in 4.7.3 (Safe Browsing), and 4.13.3 (Protect yourself from spam), I will add a paragraph in 4.12 in a future revision of my guide]_.

3) Firefox relay and Proton Pass are similar. 

4) Anti-spam protection is partial: it is based on "if your email address is not known, it cannot be spammed". But that's wrong: spamming servers spend their time to send mails with generated usernames; when they receive no error, the username exist.

Finally, I have just mentioned DuckDuckGo, Firefox and Proton alias in anti-spam techniques, though I prefer disposable addresses.

Regards,

MN

**Comment by Jopp-gh** [comment]
I'm not sure if here's the right place to ask of some questionable "security update" software of last week. Security isn't easy to research and then to implement. Thanks very much for your hard work and to share this "hot iron" with us so generously.
@MikeNavy

**Comment by AtomicRobotMan0101** [comment]
Id completely forgotten this thread.

I just re-grabbed the doco from the link.

The fact the OP has updated it is very nice indeed.  (2024-05-19 - Revision 27)

**Comment by MikeNavy** [comment]
Hi,

@Evan McBeth @Jopp-gh

Thank you for your support.

You can find a similar thread on Linux Mint forum,
https://forums.linuxmint.com/viewtopic.php?t=397740

This thread is a bit more detailed, you will find the list of changes
for each revision.

Regards,

MN

Le 02/06/2024 à 02:54, Evan McBeth a écrit :

**Reply by 100savage** [reply]
Thank you for your guide. I created a small app based off of your 4.7 Safe browsing section. My app is located here: https://github.com/100savage/Savage-Blocker

**Reply by MikeNavy** [reply]
Hi,
There is a choice to make between a rich "/etc/hosts" file and a reduced hosts file and a rich uBlock Origin:
- The IP addresses you block in the hosts file work "system wide", for any application and any connection; that's good. However, you can't disable them for some reliable websites that could not work correctly with the filters.
- At the opposite, the filters you put in uBlock Origin work only on the browser. But with uBlock Origin you can disable the filtering for a complete website, or for a given web page, or even continue browsing for once when a filter prevents it.

The problem is the same if you use a filtering DNS (see §4.11 page 49) such as Mullvad ones: you can't disable a filter for a given website. I do use Mullvad "base.dns.mullvad.net" as a private DNS on my Android smartphone, but on my Linux Mint computer, that I use more than my smartphone, I use another filtering DNS "zero.dns0.eu", security oriented, that mainly filters phishing websites.

Same problem finally with some extensions: ClearURLs is a nice anti-tracking extension, but I prefer to use "ClearURLs for uBo" filters list in UBlock Origin, since I can disable it when I disable uBlock Origin.

So, my compromise for my Linux Mint computer is to use "/etc/hosts" to filter malware, spam, and crypto miners, and to use a "light" filtering DNS; I use uBlock Origin for anti-tracking, privacy, ads and some more malware filter lists not available as hosts format but as AdBlock one. Particularly, the use of Steven Black hosts list in the hosts file would make a lot of websites to dysfunction; if you want to block some ads in the hosts file, just use Yoyo adservers block list, lighter than Steven Black's one,
https://pgl.yoyo.org/adservers/serverlist.php?hostformat=hosts&showintro=0&mimetype=plaintext

I will mention your Savage-Blocker in a future revision of my guide (I prepare a paragraph about private / anonymous messaging).

Regards,

MN
